g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/source/crons/SourceSendingCron.java

```diff
@@ -367,45 +343,77 @@ public void syncBkSourceToRemote() {
 			return;
 		}
 		
-		if (!remoteGRMURL.equalsIgnoreCase(LOCAL_STR) && !grmConnected) {
+		if (!remoteGRMURL.equalsIgnoreCase(LOCAL_STR)) {
 			try {
 				remoteSyncService.ping(remoteGRMURL);
+				setGrmConnected(true);
 				List<SyncRecordModel> syncGrmList = sqlLite.getSynRecords(grmFlag);
 				if(syncGrmList != null) {
 					for(SyncRecordModel syncGrm : syncGrmList) {
 						ComponentSourceDTO comDTO = getCachedLocalBundle(syncGrm);
-						remoteSyncService.send(comDTO, remoteGRMURL);
-						
+                        try {
+						   remoteSyncService.send(comDTO, remoteGRMURL);
+                        } catch (L10nAPIException e) {
+                        	sendKeySource2GRM(comDTO, remoteGRMURL);
+            			}
+                        remoteSyncService.ping(remoteGRMURL);
```

Comment:
line 359: is the ping necessary?

Reply:
Judge whether L10nAPIException is network exception or GRM request body limit issue